File: Sources/Swift/FastText.swift
//
//  FastText.swift
//
//
//  Created by Noah Little on 22/1/2024.
//

import Foundation
@_exported import ObjCFastText

final class FastText {
    private let wrapper: FastTextWrapper
    
    init(modelPath: URL) {
        self.wrapper = .init(model: modelPath)
    }
    
    func nearbyWords(word: String) -> [String] {
        var arr = NSMutableArray()
        wrapper.nearbyWords(arr, forWord: word)
        
        return [""]
    }
}

File: Sources/OBJC/FastText.cpp
//
//  FastText.cpp
//  
//
//  Created by Noah Little on 22/1/2024.
//

#ifdef __cplusplus

#include <stdio.h>
#include "include/FastText.h"

using namespace fasttext;

fasttext::FastText * fastTextMain;

void fastTextInit(const std::string& model_path) {
    if (fastTextMain == NULL) {
        fastTextMain = new FastText();
        fastTextMain->loadModel(model_path);
    }
}

void fasttextGetNearByWords(const char* word,
                            std::vector<std::pair<real, std::string>> &nearestNeighbours) {
    nearestNeighbours = fastTextMain->getNN(word, 10);
}

#endif
